marimo/_convert/non_marimo_python_script.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Marimo. All rights reserved.
+"""Convert non-marimo Python scripts to marimo notebooks."""
+
+from __future__ import annotations
+
+import json
+
+from marimo._convert.ipynb import convert_from_ipynb_to_notebook_ir
+from marimo._schemas.serialization import NotebookSerialization
+
+
+def convert_non_marimo_python_script_to_notebook_ir(
+    source: str,
+) -> NotebookSerialization:
+    """Convert a Python script that isn't in the notebook format into marimo notebook IR.
+
+    This should only be called after verifying the file is not already
+    a valid marimo notebook. It converts by:
+    1. If pypercent format is detected (# %%), use jupytext for conversion
+    2. Otherwise, puts all content into a single cell
+    3. Preserves PEP 723 inline script metadata if present
+    """
+    if "# %%" in source:
+        try:
+            import jupytext  # type: ignore[import-untyped]
+        except ImportError as e:
+            raise ImportError(
+                "Converting py:percent format requires jupytext"
+            ) from e
+        notebook = jupytext.reads(source, fmt="py:percent")
+    else:
+        notebook = {"cells": [{"source": source, "cell_type": "code"}]}
+
+    notebook_str = json.dumps(notebook)
+
+    ir = convert_from_ipynb_to_notebook_ir(notebook_str)
```

Comment:
this would require `nbformat` is cases where you probably don't really need it.

it might be easier to just create the `ir` in the `else` block

Reply:
ohh nice, good call